src/Object/Search/ContactResultList.php

```diff
@@ -5,11 +5,11 @@
 use Friendica\Model\Search;
 
 /**
- * A list of search results with details
+ * A list of search results for contacts with metadata
  *
  * @see Search for details
  */
-class ResultList
+class ContactResultList
```

Comment:
Is this class specific to contacts search results? Couldn't it be used to collect post search results?

Reply:
I like the way this goes!

```php
class ResultList
{
   /**
      * @var IResult
    **/
    private $results;
}

interface IResult
{
}

class ContactResult implements IResult
{
}
```

Reply:
I tend to favor inheritance over interface. Parent classes can define default values while Interface can't, for example. In this case:
```php
class ResultList
{
   /**
    * @var Result[]
    **/
    private $results;
}

abstract class Result
{
}

class ContactResult extends Result
{
}
```

Reply:
Hm.. I normally prefer interfaces because of more decoupling and optionally add a abstract class to define abstract methods. So you don't have to inheritate the whole class if you don't need to, but it's possible.
Another reason is that inheritance could possibly increase the complexity too because you possible have some business logic on other places than in the class itself. Here I'm thinking about coders who aren't so familar with inheritance concepts.

Reply:
Do what you prefer, I presented my own case, both are valid, it seems to boil down to personal preference at this point.